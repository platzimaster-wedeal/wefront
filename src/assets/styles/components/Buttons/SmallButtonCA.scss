@import '../../partials/colors.scss';
@import '../../partials/shadows.scss';
@import "../../Breakpoints.scss";

.button-actionCA {
    background-color: $color-background;
    border-radius: 50%;
    display: flex;
    justify-content: center;
    align-items: center;
    color: $color-primary;
    box-shadow: 0 0 12px 6px rgba(0,0,0,.03);
    width: 20px;
    height: 20px;
    font-size: 15px;
    cursor: pointer;
    
    @include media(mobile-l) {
        width: 30px;
        height: 30px;
        font-size: 18px;
    }
    @include media(tablet) {
        width: 42px;
        height: 42px;
        font-size: 30px;
    }
}