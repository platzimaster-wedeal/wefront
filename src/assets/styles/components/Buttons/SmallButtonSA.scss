@import '../../partials/colors.scss';
@import '../../partials/shadows.scss';
@import "../../breakpoints.scss";


.button-actionSA {
    background-color: $color-background;
    border-radius: 15%;
    display: flex;
    justify-content: center;
    align-items: center;
    color: $color-primary;
    box-shadow: $shadow-small;
    width: 20px;
    height: 20px;
    font-size: 15px;
    cursor: pointer;
    
    @include media(mobile-l) {
        width: 24px;
        height: 24px;
        font-size: 20px;
    }
    @include media(tablet) {
        width: 42px;
        height: 42px;
        font-size: 30px;
    }
}