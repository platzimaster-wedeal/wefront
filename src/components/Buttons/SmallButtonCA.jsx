import React from "react";
import "../../assets/styles/components/Buttons/SmallButtonCA.scss";

const buttonSize = (closeBtn) => {
 if (closeBtn) {
  return {
   backgroundColor: " hsl(152,59%,94%)",
   color: " hsla(240,27%,12%,.6)",
  };
 }
};

const SmallButtonCA = ({ icon, closeBtn, className, onClick }) => {
 const MdIcon = icon;
 return (
  <button
   className={`button-actionCA ${className}`}
   style={buttonSize(closeBtn)}
   onClick={onClick}>
   <MdIcon />
  </button>
 );
};

export default SmallButtonCA;
