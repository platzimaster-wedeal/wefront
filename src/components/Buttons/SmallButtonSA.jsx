import React from "react";
import "../../assets/styles/components/Buttons/SmallButtonSA.scss";

const SmallButtonSA = ({ icon, className }) => {
 const MdIcon = icon;
 return (
  <button className={`button-actionSA ${className}`}>
   <MdIcon />
  </button>
 );
};

export default SmallButtonSA;
