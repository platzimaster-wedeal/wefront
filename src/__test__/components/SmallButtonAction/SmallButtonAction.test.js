import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import { MdAdd } from "react-icons/md";
import SmallButtonSA from "../../../components/Buttons/SmallButtonSA";
import SmallButtonCA from "../../../components/Buttons/SmallButtonCA";

describe("<SmallButtonSA/>", () => {
 const buttonSA = mount(<SmallButtonSA icon={MdAdd} />);
 test("SmallButtonSA render test", () => {
  expect(buttonSA.length).toEqual(1);
 });
 test("UI Snapchot for SmallButtonAction", () => {
  const squareButtonAction = create(<SmallButtonSA icon={MdAdd} />);
  expect(squareButtonAction.toJSON()).toMatchSnapshot();
 });
});

describe("<SmallButtonCA/>", () => {
 const buttonCA = mount(<SmallButtonCA icon={MdAdd} />);
 test("SmallButtonCA render test", () => {
  expect(buttonCA.length).toEqual(1);
 });
 test("UI Snapshot for SmallButtonCA component", () => {
  const circleButtonAction = create(<SmallButtonCA icon={MdAdd} />);
  expect(circleButtonAction.toJSON()).toMatchSnapshot();
 });
});
