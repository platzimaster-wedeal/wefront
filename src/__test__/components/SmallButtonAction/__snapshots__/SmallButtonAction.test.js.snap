// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmallButtonCA/> UI Snapshot for SmallButtonCA component 1`] = `
<button
  className="button-actionCA undefined"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
    />
  </svg>
</button>
`;

exports[`<SmallButtonSA/> UI Snapchot for SmallButtonAction 1`] = `
<button
  className="button-actionSA undefined"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
    />
  </svg>
</button>
`;
